app/questionnaire/questionnaire_schema.py

```diff
@@ -855,9 +855,67 @@ def _populate_when_rules_section_dependencies(self) -> None:
                     section["id"]
                 ] = rules_section_dependencies
 
+            if rules_section_dependents := self._get_rules_section_dependents(
+                section["id"], rules
+            ):
+                for dependent in rules_section_dependents:
+                    if (
+                        dependent
+                        not in self._when_rules_section_dependencies_by_section
+                    ):
+                        self._when_rules_section_dependencies_by_section[
+                            dependent
+                        ] = set()
+
+                    self._when_rules_section_dependencies_by_section[dependent].add(
+                        section["id"]
+                    )
+
+    def _get_rules_section_dependents(
+        self, current_section_id: str, rules: Union[Mapping, Sequence]
```

Comment:
Minor but we can update to follow the new Type hint guidance now for things like Union and Optional